api/billingd/service/service.go

```diff
@@ -806,10 +809,18 @@ func (s *Service) handleUsage(ctx context.Context, cus *Customer, product Produc
 		log.Warnf("negative %s detected: total=%d inc=%d)", product.Key, total, incSize)
 		total = 0
 	}
+	update := bson.M{"daily_usage." + product.Key + ".total": total}
 	if total > product.FreeQuotaSize && !cus.Billable {
-		return nil, common.ErrExceedsFreeQuota
+		now := time.Now().Unix()
+		if cus.GracePeriodStartAt == 0 {
+			cus.GracePeriodStartAt = now
+			update["grace_period_start_at"] = cus.GracePeriodStartAt
+		}
+		deadline := cus.GracePeriodStartAt + int64(s.config.FreeQuotaGracePeriod.Seconds())
+		if now >= deadline {
+			return nil, common.ErrExceedsFreeQuota
+		}
```

Comment:
Grace period starts when the user initially goes over the limit. This does not account for the case where they might drop down again. Could add that later, but this is good nuff for now. Also allows for a `0` grace period.